frontends/open-discussions/package.json

```diff
@@ -1,5 +1,8 @@
 {
   "name": "open-discussions",
+  "installConfig": {
+    "hoistingLimits": "workspaces"
+  },
```

Comment:
Hoisting is a mechanism for deduplicating dependencies in a dependency tree. See [hoisting](https://classic.yarnpkg.com/blog/2018/02/15/nohoist/) for more background. *That article is from Yarn 1; we use Yarn 3, but the concepts are the same.*

Hoisting can be problematic, particularly with peer dependencies. For example:
- workspace `open-discussions` (old open) uses React 16 + Enzyme for testing, with React 16 as a peer dependency
- workspace `infinite-corridor` is using React 18 + React Testing Library (v14) for testing, with React 18 as a peer dependency

If `@testing-library/react` gets hoisted to root node_modules, it will use the version of React in root node_modules. But is that 16 (from open-discussions) or 18 (from infinite-corridor). Not super predictable. Bad.

```
git repo
|-- package.json
|-- node_modules                A (root)
|-- frontends
    |-- infinite_corridor
        |-- package.json        <-- specifies React 18
        |-- node_mudles         B      
    |-- open_discussions
        |-- package.json        <-- specifies React 16
        |-- node_modules        C
    |-- ...others

Does React 16 go in A or C?
           18 go in A or B?

Not easily predictable.
```

### Enter `hoistingLimits`
Since workspace `open-discussions` is separate from all the other workspaces, the simplest solution is to never hoist its dependencies to root. So in the tree above, all dependencies from `open-discussions/package.json` end up in `open-discussions/node_modules`. 

It's still possible for workspace `open-discussions` to use dependencies from root `node_modules` if open-discussions is using a dependency without specifying it in the package.json. This was the case with `immer`:
- `immer` was previously specified in the root package.json...
-  `immer` was being used by workspace `open-discussions`. *Because of Node's module resolution ("keep searching in higher and higher node_modules"), that was working fine.*

So to really isolate open-discussions, we want:
1. do not hoist its dependencies to root node_modules
2. make sure it's package.json specifies its dependencies

(2) is taken care of by a linting rule.